drivers/ethernet/phy_sam_gmac.c

```diff
@@ -187,6 +187,22 @@ u32_t phy_sam_gmac_id_get(const struct phy_sam_gmac_dev *phy)
 	return phy_id;
 }
 
+bool phy_sam_gmac_link_status_get(const struct phy_sam_gmac_dev *phy)
+{
+	Gmac *const gmac = phy->regs;
+	u32_t bmsr;
+
+	mdio_bus_enable(gmac);
```

Comment:
Don't really understand the reason behind enabling / disabling MDIO interface on every PHY R/W transaction. Does it save power or something ? If not, we should probably just enable MDIO when initializing the device and leave it "as is" forever.. This will reduce the lines of code..

Reply:
For this change, I simply replicated what other functions are doing (e.g. `phy_sam_gmac_id_get`).

If this is unnecessary, then we will need to fix all the other functions as well.

Reply:
I can see that the ASF implementation does this as well. At this point, we can just say- rationale: ASF.

Reply:
I'm fine with leaving it as-is or fixing it in the future. I was more interested in knowing if there is a strong rationale to do it this way because I always enable the MDIO interface when the device is booting up, and don't disable it ever again.